src/ConstructiveSolidGeometry/LinePrimitives/Edge.jl

```diff
@@ -1,8 +1,29 @@
-struct Edge{T} <: AbstractLinePrimitive{T}
+struct Edge{T<:AbstractFloat} <: AbstractLinePrimitive{T}
     a::CartesianPoint{T}
     b::CartesianPoint{T}
 end
 
+#Type promotion happens here
+function Edge(a::A, b::B) where {A<:(CartesianPoint), B<:(CartesianPoint)}
```

Comment:
I don't think we need brackets here around the types `PT <: (CartesianPoint), ...` -> `PT <: CartesianPoint, ...`